allennlp/modules/token_embedders/openai_transformer_embedder.py

```diff
@@ -0,0 +1,110 @@
+from typing import List
+import torch
+
+from allennlp.modules import ScalarMix
+from allennlp.modules.token_embedders.token_embedder import TokenEmbedder
+from allennlp.modules.openai_transformer import OpenaiTransformer
+
+
+@TokenEmbedder.register("openai_transformer_embedder")
+class OpenaiTransformerEmbedder(TokenEmbedder):
+    """
+    Takes a byte-pair representation of a batch of sentences
+    (as produced by the ``OpenaiTransformerBytePairIndexer``)
+    and generates the corresponding contextual embeddings.
+
+    Parameters
+    ----------
+    transformer: ``OpenaiTransformer``, required.
+        The ``OpenaiTransformer`` module used for the embeddings.
+    num_output_representations: ``int``, optional (default: 1)
+        How many "scalar mixes" of the embedding layers to return.
+        (Currently only implemented for n = 1.)
+    do_layer_norm: ``bool`` (default: ``False``)
+        Whether the ``ScalarMix``es should use layer norm.
+    """
+    def __init__(self,
+                 transformer: OpenaiTransformer,
+                 num_output_representations: int = 1,
+                 do_layer_norm: bool = False) -> None:
+        super().__init__()
+
+        if num_output_representations > 1:
+            raise NotImplementedError("more than 1 output representation is not implemented")
+
+        self._transformer = transformer
+        self._num_output_representations = num_output_representations
+        self._do_layer_norm = do_layer_norm
+
+        self._scalar_mixes: List[ScalarMix] = []
```

Comment:
We only need one `ScalarMix` since `num_output_representations==1`.

Reply:
done